plugins/woocommerce/src/RestApi/Routes/V4/OrderNotes/Controller.php

```diff
@@ -137,53 +168,39 @@ protected function prepare_links( $item, WP_REST_Request $request ): array {
 	 * @return array
 	 */
 	protected function get_item_response( $note, WP_REST_Request $request ): array {
-		return array(
-			'id'               => (int) $note->comment_ID,
-			'order_id'         => (int) $note->comment_post_ID,
-			'author'           => $note->comment_author,
-			'date_created'     => wc_rest_prepare_date_response( $note->comment_date ),
-			'date_created_gmt' => wc_rest_prepare_date_response( $note->comment_date_gmt ),
-			'note'             => $note->comment_content,
-			'is_customer_note' => (bool) get_comment_meta( $note->comment_ID, 'is_customer_note', true ),
-		);
+		return $this->item_schema->get_item_response( $note, $request );
 	}
 
 	/**
-	 * Check if a given request has access to the order.
+	 * Check if a given request has access to read an item.
 	 *
-	 * @param  WC_Order|boolean $order The order object.
+	 * @param  WP_REST_Request $request The request object.
 	 * @return WP_Error|boolean
 	 */
-	protected function order_permissions_check( $order ) {
-		if ( ! $order || ! $order instanceof WC_Order ) {
-			return $this->get_route_error_response( $this->get_error_prefix() . 'invalid_id', __( 'Invalid order ID.', 'woocommerce' ), WP_Http::NOT_FOUND );
-		}
+	public function get_item_permissions_check( $request ) {
+		$order = $this->get_order_by_note_id( (int) $request['id'] );
 
-		if ( ! $this->check_permissions( 'order', 'edit', (int) $order->get_id() ) ) {
-			return $this->get_route_error_response( $this->get_error_prefix() . 'cannot_edit', __( 'Sorry, you are not allowed to access notes for this order.', 'woocommerce' ), rest_authorization_required_code() );
+		if ( ! $order ) {
+			return $this->get_route_error_by_code( self::INVALID_ID );
 		}
 
+		if ( ! wc_rest_check_post_permissions( 'shop_order', 'read', $order->get_id() ) ) {
+			return $this->get_authentication_error_by_method( $request->get_method() );
+		}
 		return true;
 	}
 
-	/**
-	 * Check if a given request has access to read an item.
-	 *
-	 * @param  WP_REST_Request $request The request object.
-	 * @return WP_Error|boolean
-	 */
-	public function get_item_permissions_check( $request ) {
-		return $this->order_permissions_check( Utils::get_order_by_note_id( (int) $request['id'] ) );
-	}
-
 	/**
 	 * Check if a given request has access to read items.
 	 *
 	 * @param  WP_REST_Request $request Full details about the request.
 	 * @return WP_Error|boolean
 	 */
 	public function get_items_permissions_check( $request ) {
-		return $this->order_permissions_check( Utils::get_order_by_id( (int) $request['order_id'] ) );
+		if ( ! wc_rest_check_post_permissions( 'shop_order', 'read', $request['order_id'] ) ) {
```

Comment:
I've noticed that we cast `$request['id']` to `(int)` above; should we do the same for `order_id` here?